### PR TITLE
[5.2] Make trans_choice receive arrays and countable objects

### PR DESCRIPTION
**Pull request by mlanin**
`trans_choice` function or `Translator::transChoice` method can receive only integers to make a proper translation selection. But when you are using this functions, most of the time you are using them with collections and arrays, so you have to always surround your variables with count() like this:

`{{ trans_choice('pages', count($pages)) }}`

This PR helps this functions to receive arrays and `Countable` objects too and they will count elements for you.

### CODE REVIEW

File: src/Illuminate/Foundation/helpers.php [hunk 1]
```diff
@@ -664,7 +664,7 @@ function trans($id = null, $parameters = [], $domain = 'messages', $locale = nul
      * Translates the given message based on a count.
      *
      * @param  string  $id
-     * @param  int     $number
+     * @param  int|array|Countable  $number
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/helpers.php]: missing leading slash

File: src/Illuminate/Translation/Translator.php [hunk 2]
```diff
@@ -184,7 +185,7 @@ protected function sortReplacements(array $replace)
      * Get a translation according to an integer value.
      *
      * @param  string  $key
-     * @param  int     $number
+     * @param  int|array|Countable  $number
```
Comment (GrahamCampbell) [inline src/Illuminate/Translation/Translator.php]: same

File: src/Illuminate/Translation/Translator.php [hunk 3]
```diff
@@ -184,7 +185,7 @@ protected function sortReplacements(array $replace)
      * Get a translation according to an integer value.
      *
      * @param  string  $key
-     * @param  int     $number
```
Comment (GrahamCampbell) [inline src/Illuminate/Translation/Translator.php]: You need to update the contract too.

Comment (mlanin) [inline src/Illuminate/Translation/Translator.php]: There was a `use` statement for Countable. So I removed trailing slash. Returned it back.

File: src/Illuminate/Translation/Translator.php [hunk 4]
```diff
@@ -216,7 +220,7 @@ public function trans($id, array $parameters = [], $domain = 'messages', $locale
      * Get a translation according to an integer value.
      *
      * @param  string  $id
-     * @param  int     $number
+     * @param  int|array|\Countable  $number
```
Comment (GrahamCampbell) [inline src/Illuminate/Translation/Translator.php]: Please update our contract too.

Comment (mlanin) [inline src/Illuminate/Translation/Translator.php]: @GrahamCampbell what contract do you mean? There is no contracts for translator. Only Symfony TranslatorInterface

Comment (morloderex) [inline src/Illuminate/Translation/Translator.php]: @GrahamCampbell Does the translator have a contract?

Comment (GrahamCampbell) [inline src/Illuminate/Translation/Translator.php]: Oh, maybe it doesn't. My bad then.

**Pull request merged**